Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java

```diff
@@ -449,4 +436,35 @@ public void onVaultEditButtonClick(View view) {
 
         mBraintreeFragment.sendAnalyticsEvent("manager.appeared");
     }
+
+    private void showVaultedPaymentMethods(List<PaymentMethodNonce> paymentMethodNonces, boolean googlePayEnabled) {
+        mSupportedPaymentMethodsHeader.setText(R.string.bt_other);
+        mVaultedPaymentMethodsContainer.setVisibility(View.VISIBLE);
+
+        VaultedPaymentMethodsAdapter vaultedPaymentMethodsAdapter = new VaultedPaymentMethodsAdapter(new PaymentMethodNonceCreatedListener() {
+            @Override
+            public void onPaymentMethodNonceCreated(PaymentMethodNonce paymentMethodNonce) {
+                if (paymentMethodNonce instanceof CardNonce) {
+                    mBraintreeFragment.sendAnalyticsEvent("vaulted-card.select");
+                }
+
+                DropInActivity.this.onPaymentMethodNonceCreated(paymentMethodNonce);
+            }
+        }, paymentMethodNonces);
+
+        vaultedPaymentMethodsAdapter.setup(
+                this, mConfiguration, mDropInRequest, googlePayEnabled, mClientTokenPresent);
+        mVaultedPaymentMethodsView.setAdapter(vaultedPaymentMethodsAdapter);
+
+        if (mDropInRequest.isVaultManagerEnabled()) {
+            mVaultManagerButton.setVisibility(View.VISIBLE);
+        }
+
+        for (PaymentMethodNonce nonce : paymentMethodNonces) {
+            if (nonce instanceof CardNonce) {
+                mBraintreeFragment.sendAnalyticsEvent("vaulted-card.appear");
```

Comment:
I wonder if we should move this analytics event into `VaultedPaymentMethodsAdapter` since vaulted cards might not actually appear if cards aren't enabled by the merchant in the `DropInRequest`.

Reply:
Alternatively, we could move the construction of `AvailablePaymentMethodNonceList` into this method, and then only pass the available payment method nonces into `VaultedPaymentMethodsAdapter`. Then we would have access to the list of displayed vaulted cards in this method, and we could use that to determine whether or not to send this event.

Reply:
Yeah that analytics event is interesting we should preserve existing behavior. I think iterating over the filtered list makes a lot of sense.

Reply:
We added a check in the `AvailablePaymentMethodNonceList` to see if the list has cards and only fire this event if it does.